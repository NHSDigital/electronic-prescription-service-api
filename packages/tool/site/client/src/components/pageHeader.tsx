import * as React from "react"
import {useContext} from "react"
import {Header, Images} from "nhsuk-react-components"
import {AppContext} from "../index"
import {isSandbox} from "../services/environment"
import SessionTimer from "./sessionTimer"
import styled from "styled-components"

interface PageHeaderProps {
  loggedIn: boolean
}

const StyledImages = styled(Images)`
  background: #005eb8;
  width: 30px;
  margin-left: 25px;
`

export const PageHeader: React.FC<PageHeaderProps> = ({loggedIn}) => {
  const {baseUrl, environment} = useContext(AppContext)
  return (
    <Header transactional>
      <Header.Container>
        <Header.Logo href={baseUrl} />
        {loggedIn ? (
          <Header.ServiceName href={`${baseUrl}config`}>
            <div className="inline-flex">
              EPSAT - Electronic Prescription Service API Tool
              <StyledImages srcSet={`${baseUrl}static/Cogs_SVG_White.svg`} sizes="50px" />
            </div>
          </Header.ServiceName>
        ) : (
          <Header.ServiceName href={baseUrl}>EPSAT - Electronic Prescription Service API Tool</Header.ServiceName>
        )}
        <SessionTimer />
      </Header.Container>
      {loggedIn && (
        <Header.Nav>
          <Header.NavItem href={baseUrl}>Home</Header.NavItem>
          <Header.NavItem href={`${baseUrl}my-prescriptions`}>My Prescriptions</Header.NavItem>
          {!isSandbox(environment) && <Header.NavItem href={`${baseUrl}logout`}>Logout</Header.NavItem>}
        </Header.Nav>
      )}
    </Header>
  )
}
