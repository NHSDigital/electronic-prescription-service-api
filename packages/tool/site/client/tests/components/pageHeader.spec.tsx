import {render, waitFor} from "@testing-library/react"
import {screen} from "@testing-library/dom"
import pretty from "pretty"
import * as React from "react"
import {PageHeader} from "../../src/components/pageHeader"
import {expect} from "@jest/globals"
import {MemoryRouter} from "react-router-dom"

test("Header renders when logged in", async () => {
  const {container} = render(
    <MemoryRouter>
      <PageHeader loggedIn={true}/>
    </MemoryRouter>
  )

  expect(screen.getByText("My Prescriptions")).toBeTruthy()

  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Header renders when not logged in", async () => {
    const {container} = render(
      <MemoryRouter>
        <PageHeader loggedIn={false}/>
      </MemoryRouter>
    )
  
    expect(screen.queryByText("My Prescriptions")).toBeNull()
  
    expect(pretty(container.innerHTML)).toMatchSnapshot()
})
  
