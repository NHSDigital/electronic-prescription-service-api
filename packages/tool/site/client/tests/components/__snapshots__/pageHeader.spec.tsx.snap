// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders when logged in 1`] = `
"<header class="nhsuk-header nhsuk-header__transactional" role="banner">
  <div class="nhsuk-width-container nhsuk-header__container">
    <div class="nhsuk-header__logo nhsuk-header__transactional--logo"><a class="nhsuk-header__link" aria-label="NHS homepage" href="/"><svg class="nhsuk-logo" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 16" height="40" width="100" aria-labelledby="nhsuk-logo_title">
          <title id="nhsuk-logo_title">NHS Logo</title>
          <path class="nhsuk-logo__background" d="M0 0h40v16H0z" fill="#005eb8"></path>
          <path class="nhsuk-logo__text" fill="#fff" d="M3.9 1.5h4.4l2.6 9h.1l1.8-9h3.3l-2.8 13H9l-2.7-9h-.1l-1.8 9H1.1M17.3 1.5h3.6l-1 4.9h4L25 1.5h3.5l-2.7 13h-3.5l1.1-5.6h-4.1l-1.2 5.6h-3.4M37.7 4.4c-.7-.3-1.6-.6-2.9-.6-1.4 0-2.5.2-2.5 1.3 0 1.8 5.1 1.2 5.1 5.1 0 3.6-3.3 4.5-6.4 4.5-1.3 0-2.9-.3-4-.7l.8-2.7c.7.4 2.1.7 3.2.7s2.8-.2 2.8-1.5c0-2.1-5.1-1.3-5.1-5 0-3.4 2.9-4.4 5.8-4.4 1.6 0 3.1.2 4 .6"></path>
        </svg></a></div>
    <div class="nhsuk-header__transactional-service-name"><a class="nhsuk-header__transactional-service-name--link" href="/config">
        <div class="inline-flex">EPSAT - Electronic Prescription Service API Tool<figure class="nhsuk-image"><img class="nhsuk-image__img sc-kAFUCS cVjqtx" srcset="/static/Cogs_SVG_White.svg" sizes="50px"></figure>
        </div>
      </a></div>
  </div>
  <div class="nhsuk-navigation-container">
    <nav class="nhsuk-navigation" id="header-navigation" role="navigation">
      <ul class="nhsuk-header__navigation-list nhsuk-header__navigation-list--left-aligned">
        <li class="nhsuk-header__navigation-item"><a class="nhsuk-header__navigation-link" href="/">Home</a></li>
        <li class="nhsuk-header__navigation-item"><a class="nhsuk-header__navigation-link" href="/my-prescriptions">My Prescriptions</a></li>
        <li class="nhsuk-header__navigation-item"><a class="nhsuk-header__navigation-link" href="/logout">Logout</a></li>
      </ul>
    </nav>
  </div>
</header>"
`;

exports[`Header renders when not logged in 1`] = `
"<header class="nhsuk-header nhsuk-header__transactional" role="banner">
  <div class="nhsuk-width-container nhsuk-header__container">
    <div class="nhsuk-header__logo nhsuk-header__transactional--logo"><a class="nhsuk-header__link" aria-label="NHS homepage" href="/"><svg class="nhsuk-logo" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 16" height="40" width="100" aria-labelledby="nhsuk-logo_title">
          <title id="nhsuk-logo_title">NHS Logo</title>
          <path class="nhsuk-logo__background" d="M0 0h40v16H0z" fill="#005eb8"></path>
          <path class="nhsuk-logo__text" fill="#fff" d="M3.9 1.5h4.4l2.6 9h.1l1.8-9h3.3l-2.8 13H9l-2.7-9h-.1l-1.8 9H1.1M17.3 1.5h3.6l-1 4.9h4L25 1.5h3.5l-2.7 13h-3.5l1.1-5.6h-4.1l-1.2 5.6h-3.4M37.7 4.4c-.7-.3-1.6-.6-2.9-.6-1.4 0-2.5.2-2.5 1.3 0 1.8 5.1 1.2 5.1 5.1 0 3.6-3.3 4.5-6.4 4.5-1.3 0-2.9-.3-4-.7l.8-2.7c.7.4 2.1.7 3.2.7s2.8-.2 2.8-1.5c0-2.1-5.1-1.3-5.1-5 0-3.4 2.9-4.4 5.8-4.4 1.6 0 3.1.2 4 .6"></path>
        </svg></a></div>
    <div class="nhsuk-header__transactional-service-name"><a class="nhsuk-header__transactional-service-name--link" href="/">EPSAT - Electronic Prescription Service API Tool</a></div>
  </div>
</header>"
`;
